app/lib/foreman_deployments/registry.rb

```diff
@@ -0,0 +1,67 @@
+module ForemanDeployments
+  class Registry
+    class TypeException < ::Foreman::Exception; end
+
+    ALLOWED_TYPES = {
+      'task' => 'ForemanDeployments::Tasks::BaseDefinition',
+      'input' => 'ForemanDeployments::Inputs::BaseInputDefinition'
+    }
+
+    def initialize
+      clear!
+    end
+
+    def clear!
+      @available = Hash[ALLOWED_TYPES.keys.map { |k| [k, {}] }]
+    end
+
+    def available(type)
+      @available[type].clone
+    end
+
+    def register(type, registered_class)
```

Comment:
Maybe it would be better make this method private, and use some meta-definitions to define public interface?
Something like what I did here: https://github.com/theforeman/foreman/pull/2665/files#diff-ea7bb2d5eb6aaaee8d590ee61f30941cR5